airflow/providers/smtp/provider.yaml

```diff
@@ -54,3 +54,27 @@ connection-types:
 
 notifications:
   - airflow.providers.smtp.notifications.smtp.SmtpNotifier
+
+config:
+  smtp_provider:
+    description: "Options for SMTP provider."
+    options:
+      ssl_context:
+        description: |
+          ssl context to use when using SMTP and IMAP SSL connections. By default, the context is "default"
+          which sets it to ``ssl.create_default_context()`` which provides the right balance between
+          compatibility and security, it however requires that certificates in your operating system are
+          updated and that SMTP/IMAP servers of yours have valid certificates that have corresponding public
+          keys installed on your machines. You can switch it to "none" if you want to disable checking
+          of the certificates, but it is not recommended as it allows MITM (man-in-the-middle) attacks
+          if your infrastructure is not sufficiently secured. It should only be set temporarily while you
+          are fixing your certificate configuration. This can be typically done by upgrading to newer
+          version of the operating system you run Airflow components on,by upgrading/refreshing proper
+          certificates in the OS or by updating certificates for your mail servers.
+
+          If you do not set this option explicitly, it will use Airflow "email.ssl_context" configuration,
+          but if this configuration is not present, it will use "default" value.
+        type: string
+        version_added: 1.3.0
+        example: "default"
+        default: ~
```

Comment:
Why do we not wish to set default: "default" here? Looking at the description above looks like we do intend to set a default of "default".

Reply:
This is sa, ereason as above - this is a chain of config: 

1) check if "smtp_provider" / "ssl_context" is set - if yes , we use it., if not we go to point 2.
2) check if "emai" /  "ssl_context" is set - if yes - we use it, if not we fallback to "default" 

So we cannot set default on "smtp_provider" one, because then we will retrieve default in point 1 and we cannot check point 2 after.


The complexity here is that we have to handle several cases: 

1) what happens if we have Airflow 2.6 and no "emai"/ssl_context
2) what happens if ewe hav Airflow 2.7 and someone set "email" / "ssl_context" = "none" and did not set "smtp_provider" / "ssl_context"

So in other words - our direct fallback is whatever the value of "email" / "ssl_context" is and only if it is not set we use "default"

Reply:
I added a complete set of unit tests showing it :)

Reply:
Back-compat engineering as recently @dstandish named it.